--- a/client/client.go
+++ b/client/client.go
@@ -4,6 +4,7 @@ package client
 
 import (
 	context "context"
+	url "net/url"
 
 	docuseal "github.com/docusealco/docuseal-go"
 	core "github.com/docusealco/docuseal-go/core"
@@ -431,3 +432,25 @@ func (c *Client) CreateSubmission(
 	}
 	return response.Body, nil
 }
+
+// The API endpoint allows you to permanently delete a document template and all of its submissions.
+func (c *Client) PermanentlyDeleteTemplate(
+	ctx context.Context,
+	// The unique identifier of the document template.
+	id int,
+	opts ...option.RequestOption,
+) (*docuseal.TemplateArchiveResult, error) {
+	opts = append(opts, option.WithQueryParameters(url.Values{"permanently": []string{"true"}}))
+	return c.ArchiveTemplate(ctx, id, opts...)
+}
+
+// The API endpoint allows you to permanently delete a submission and all of its submitters and documents.
+func (c *Client) PermanentlyDeleteSubmission(
+	ctx context.Context,
+	// The unique identifier of the submission.
+	id int,
+	opts ...option.RequestOption,
+) (*docuseal.SubmissionArchiveResult, error) {
+	opts = append(opts, option.WithQueryParameters(url.Values{"permanently": []string{"true"}}))
+	return c.ArchiveSubmission(ctx, id, opts...)
+}
