--- a/client/client.go
+++ b/client/client.go
@@ -22,6 +22,9 @@ type Client struct {

 func NewClient(apiKey string, opts ...option.RequestOption) *Client {
 	opts = append([]option.RequestOption{option.WithAPIKey(apiKey)}, opts...)
 	options := core.NewRequestOptions(opts...)
+	if options.HTTPHeader.Get("User-Agent") == "" {
+		options.HTTPHeader.Set("User-Agent", "DocuSeal Go v1.0.0")
+	}
 	return &Client{
 		WithRawResponse: NewRawClient(options),
